json2html/jsonconv.py

```diff
@@ -31,42 +37,69 @@ def convert(self,**args):
 				self.json_input = json.dumps(self.json_input)
 		else:
 			raise Exception('Can\'t convert NULL!')
-		
-		
+
+
 		ordered_json = json.loads(self.json_input, object_pairs_hook=ordereddict.OrderedDict)
 		return self.htmlConvertor(ordered_json)
 
+	def columnHeadersFromListOfDicts(self,ordered_json):
+		if len(ordered_json) < 2:
+			return None
+		if not isinstance(ordered_json[0],dict):
+			return None
+		column_headers = ordered_json[0].keys()
+		for entry in ordered_json:
+			if not isinstance(entry,dict):
+				return None
+			if len(entry.keys()) != len(column_headers):
+				return None
+			for header in column_headers:
+				if not header in entry:
+					return None
+		return column_headers
 
+	#ordered_json needs to be a dict
 	def iterJson(self,ordered_json):
+		def markup(entry, parent_is_list=False):
+			if(isinstance(entry,unicode)):
+				return unicode(entry)
+			if(isinstance(entry,int) or isinstance(entry,float)):
+				return str(entry)
+			if(parent_is_list and isinstance(entry,list)==True):
+				#list of lists are not accepted
+				return ''
+			if(isinstance(entry,list)==True) and len(entry) == 0:
+				return ''
+			if(isinstance(entry,list)==True):
+				return '<ul><li>' + '</li><li>'.join([markup(child, parent_is_list=True) for child in entry]) + '</li></ul>'
+			if(isinstance(entry,dict)==True):
+				return self.iterJson(entry)
+			return '' #safety: don't do recursion over anything that we don't know about - iteritems() will most probably fail
+
 		global a
-		a=a+ "<table border=\"1\">" 
+		global table_attributes
+
+		table_init_markup = "<table %s>" %(table_attributes)
```

Comment:
What about of no `table_attributes` is/are not present? Default table border has been overwritten. Also when I try the simple example: 

```
_json2conv = {
	"glossary": {
		"title": "example glossary",
		"GlossDiv": {
			"title": "S",
			"GlossList": {
				"GlossEntry": {
					"ID": "SGML",
					"SortAs": "SGML",
					"GlossTerm": "Standard Generalized Markup Language",
					"Acronym": "SGML",
					"Abbrev": "ISO 8879:1986",
					"GlossDef": {
						"para": "A meta-markup language, used to create markup languages such as DocBook.",
						"GlossSeeAlso": ["GML", "XML"]
					},
					"GlossSee": "markup"
				}
			}
		}
	}
}
```

I didn't get the correct output. Please re-check what's being missing. Also, do share your json, I'm curious to see the sorting as suggested by you. It would be great to see everything working perfectly on my end. Thanks for introducing sort.